tempodb/encoding/vparquet2/block_search.go

```diff
@@ -130,18 +131,31 @@ func (b *backendBlock) Search(ctx context.Context, req *tempopb.SearchRequest, o
 	return results, nil
 }
 
-func makePipelineWithRowGroups(ctx context.Context, req *tempopb.SearchRequest, pf *parquet.File, rgs []parquet.RowGroup) pq.Iterator {
+func makePipelineWithRowGroups(ctx context.Context, req *tempopb.SearchRequest, pf *parquet.File, rgs []parquet.RowGroup, dc []backend.DedicatedColumn) pq.Iterator {
 	makeIter := makeIterFunc(ctx, rgs, pf)
 
 	// Wire up iterators
 	var resourceIters []pq.Iterator
 	var traceIters []pq.Iterator
 
+	// Dedicated column mappings
+	resourceColumnMapping := dedicatedColumnsToColumnMapping(dc, dedicatedColumnScopeResource)
```

Comment:
It looks like these end up with the same logic so maybe we don't need to split them by scope.  Are you doing this so that one scope can be prioritized over the other?  If so maybe span-level should take precedence over resource-level.   Not really concerned as this is the old search functionality, but just mentioning it.

Reply:
>  Are you doing this so that one scope can be prioritized over the other?

No, this is just because in most other places require the mapping only for one scope. I changed `dedicatedColumnsToColumnMapping()` such that it can create mappings for multiple scopes.